packages/block-editor/src/hooks/block-styles.js

```diff
@@ -18,29 +18,18 @@ import DefaultStylePicker from '../components/default-style-picker';
 
 export const withInspectorControls = createHigherOrderComponent(
 	( WrappedComponent ) => ( props ) => {
-		const { blockType, selectedBlockClientId, hasBlockStyles } = useSelect(
-			( select ) => {
-				const { getSelectedBlockClientId, getBlockName } = select(
-					'core/block-editor'
-				);
-				const { getBlockStyles } = select( 'core/blocks' );
-				const blockClientId = getSelectedBlockClientId();
-				const selectedBlockName =
-					blockClientId && getBlockName( blockClientId );
-				const blockStyles =
-					blockClientId && getBlockStyles( selectedBlockName );
+		const { name, clientId, isSelected } = props;
 
-				return {
-					blockType:
-						blockClientId && getBlockType( selectedBlockName ),
-					selectedBlockClientId: blockClientId,
-					hasBlockStyles: blockStyles && blockStyles.length > 0,
-				};
-			},
-			[]
+		const hasBlockStyles = useSelect(
+			( select ) => select( blocksStore ).getBlockStyles( name ),
```

Comment:
Might be better if we calculate the boolean derived value here. One might expect `hasBlockStyles` to be boolean, based on the naming convention.

Something like this:
```js
( select ) => select( blocksStore ).getBlockStyles( name ).length > 0
```

P.S. Sorry for nitpicking.

Reply:
Don't say sorry at all :-). Sounds great to me. Going to update the line below, too.

Reply:
handled here https://github.com/WordPress/gutenberg/pull/23993/commits/acf9861a85be5be270521136a96dac0ce11e7dac